EventView-1/src/main/java/com/eventview/controller/EventTypesRestController.java

```diff
@@ -20,80 +21,62 @@
         }
         log.info("eventTypes received");
-        return new ResponseEntity<List<EvenTypes>>(evenTypes, HttpStatus.OK);
+        return new ResponseEntity<>(evenTypes, HttpStatus.OK);
     }
 
-    @GetMapping(path = "/type/{eventtypeid}")
-    public ResponseEntity<EvenTypes> findByEventtypeId(@PathVariable Integer eventtypeid) {
-        log.info("getting eventtype by eventtype_id{}", eventtypeid);
-        EvenTypes evenTypes = eventTypeService.findByEventtypeId(eventtypeid);
+    @GetMapping(path = "/event/type/{eventTypeId}")
+    public ResponseEntity<EvenTypes> findByEventTypeId(@PathVariable Integer eventTypeId) {
+        log.info("getting eventType by eventTypeId{}", eventTypeId);
+        EvenTypes evenTypes = eventTypeService.findByEventTypeId(eventTypeId);
 
-        if (evenTypes == null) {
-            log.info("eventtype with the id{} doesn't exist", eventtypeid);
-            return new ResponseEntity<EvenTypes>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<EvenTypes>(evenTypes, HttpStatus.OK);
+        return new ResponseEntity<>(evenTypes, HttpStatus.OK);
     }
 
 
-    @PostMapping(path = "/type")
+    @PostMapping(path = "/event/type")
     public ResponseEntity<EvenTypes>
-    createEventType(@RequestBody EvenTypes eventtype) {
-        log.info("creating new eventtype:{}", eventtype);
-
-        /*
-        if (eventTypeService.exists(eventtype)) {
-            log.info("eventtype with same id " + eventtype.getEventTypeId() + " exists");
-            return new ResponseEntity<EvenTypes>(HttpStatus.CONFLICT);
-        }
+    createEventType(@RequestBody EvenTypes evenTypes) {
+        log.info("creating new evenTypes:{}", evenTypes);
+        if (eventTypeService.exists(evenTypes)) throw new EventTypeExistsException("EventType already exists");
 
-         */
-
-        eventTypeService.createEventType(eventtype);
-        log.info("eventtype created");
-        return new ResponseEntity<EvenTypes>(HttpStatus.CREATED);
+        eventTypeService.createEventType(evenTypes);
+        log.info("evenTypes created");
+        return new ResponseEntity<>(HttpStatus.CREATED);
     }
 
-    @PostMapping(path = "/type/{eventtypeid}")
+    @PutMapping(path = "/event/type/{eventTypeId}")
     public ResponseEntity<Void>
-    updateEventType(@PathVariable Integer eventtypeid, @RequestBody EvenTypes evenTypes) {
-        log.info("updating eventtype:{}", evenTypes);
-        EvenTypes evenTypes1 = eventTypeService.findByEventtypeId(eventtypeid);
+    updateEventType(@PathVariable Integer eventTypeId, @RequestBody EvenTypes evenTypes) {
+        log.info("updating eventType:{}", evenTypes);
+        EvenTypes evenTypes1 = eventTypeService.findByEventTypeId(eventTypeId);
 
-        if (evenTypes1 == null) {
-            log.info("eventtype with id{} doesn't exist", eventtypeid);
-            return new ResponseEntity<Void>(HttpStatus.NOT_FOUND);
+        if (evenTypes1 != null) {
```

Comment:
what if null

Reply:
actually if condition is not necessary here.
i'm using findByEventTypeId to get the evenTypes1 so if there's no eventType with the id exception is taken care by the findByEventTypeId.
i'm removing if condition